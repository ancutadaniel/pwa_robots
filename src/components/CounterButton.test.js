import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CounterButton from './CounterButton';

it('expect to render CounterButton component', () => {
  const mockColor = 'red';

  const wrapper = shallow(<CounterButton color={mockColor} />);
  expect(shallowToJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
});

it('increments the counter correctly', () => {
  const mockColor = 'red';
  const wrapper = shallow(<CounterButton color={mockColor} />);

  wrapper.find('[id="counter"]').simulate('click');
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 2 });
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 3 });
  expect(wrapper.props().color).toEqual('red');
});
