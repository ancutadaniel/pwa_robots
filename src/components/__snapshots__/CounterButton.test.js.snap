// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CounterButton component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  UpdateCount: 
  0
</button>
`;
